### PR TITLE
Flink: make FLIP-27 default in SQL and mark the old FlinkSource as deprecated

### PR DESCRIPTION
**Pull request by stevenzwu**
SQL config default change is only applied to 1.20.

See the dev ML discussion [here](https://lists.apache.org/api/plain?thread=2r34z5drgkn1fqbvktwfzhr0fj39p3th).

### CODE REVIEW

File: flink/v1.20/flink/src/main/java/org/apache/iceberg/flink/FlinkConfigOptions.java [hunk 1]
```diff
@@ -88,7 +88,7 @@ private FlinkConfigOptions() {}
   public static final ConfigOption<Boolean> TABLE_EXEC_ICEBERG_USE_FLIP27_SOURCE =
       ConfigOptions.key("table.exec.iceberg.use-flip27-source")
           .booleanType()
-          .defaultValue(false)
+          .defaultValue(true)
```
Comment (stevenzwu) [inline flink/v1.20/flink/src/main/java/org/apache/iceberg/flink/FlinkConfigOptions.java:91]: this is the default config change. only applied to 1.20

File: flink/v1.20/flink/src/main/java/org/apache/iceberg/flink/source/IcebergSource.java [hunk 2]
```diff
@@ -86,7 +85,6 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-@Experimental
```
Comment (pvary) [inline flink/v1.20/flink/src/main/java/org/apache/iceberg/flink/source/IcebergSource.java:89]: Do we want to say that this is `PublicEvolving` instead?

**Review by pvary — APPROVED** [review]

File: flink/v1.20/flink/src/main/java/org/apache/iceberg/flink/source/IcebergSource.java (hunk 2 above)
Comment (stevenzwu) [reply flink/v1.20/flink/src/main/java/org/apache/iceberg/flink/source/IcebergSource.java:89]: I feel it is fine to be just public.

**Comment by stevenzwu** [comment]
thanks @pvary for the review

**Pull request merged**